test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,263 @@
+      forms = forms[0];
+    }
+    const beacon = new Beacon($sandbox);
+
+    return {
+      router: router,
+      form: forms,
+      beacon: beacon,
+      ac: authClient,
+      setNextResponse: setNextResponse,
+      successSpy: successSpy,
+      afterErrorHandler: afterErrorHandler,
+    };
+  }
+
+  const setupClaimsProviderFactorWithIntrospect = _.partial(setup, resAllFactors, {
+    factorType: 'claims_provider',
+    provider: 'CUSTOM',
+  });
+
+  Expect.describe('Claims Provider Factor', function() {
+    Expect.describe('when SKIP_IDP_FACTOR_VERIFICATION_BUTTON FF is on', function() {
+      itp('hides the button bar when lastFailedChallengeFactorData is null', function() {
+        return setupClaimsProviderFactorWithIntrospect({ 'features.skipIdpFactorVerificationBtn': true }, null).then(function(test) {
+          Expect.isNotVisible(test.form.buttonBar());
+        });
+      });
+
+      itp('shows the button bar when lastFailedChallengeFactorData has value', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': true }, {
+          errorMessage: 'lastFailedChallengeFactorData error message',
+        }).then(function(test) {
+          expect(test.form.buttonBar().hasClass('hide')).toBe(false);
+        });
+      });
+
+      itp('shows the waiting spinner when lastFailedChallengeFactorData is null', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': true }, null).then(function(test) {
+          expect(test.form.hasSpinner().length).toBe(1);
+        });
+      });
+
+      itp('hides the waiting spinner when lastFailedChallengeFactorData has value', function() {
+        return setupClaimsProviderFactorWithIntrospect( { 'features.skipIdpFactorVerificationBtn': true }, {
+          errorMessage: 'lastFailedChallengeFactorData error message',
+        }).then(function(test) {
+          expect(test.form.hasSpinner().length).toBe(0);
+        });
+      });
+
+      itp('shows the right auto redirect subtitle when lastFailedChallengeFactorData is null', function() {
+        return setupClaimsProviderFactorWithIntrospect({ 'features.skipIdpFactorVerificationBtn': true }, null).then(function(test) {
+          expectSubtitleToBe(test, 'Redirecting to IDP factor...');
+        });
+      });
+
+      itp('shows the right auto redirect subtitle when lastFailedChallengeFactorData has value', function() {
+        return setupClaimsProviderFactorWithIntrospect({ 'features.skipIdpFactorVerificationBtn': true }, {
+          errorMessage: 'lastFailedChallengeFactorData error message',
+        }).then(function(test) {
+          expectSubtitleToBe(test, 'Clicking below will redirect to verification with IDP factor');
+        });
+      });
+
+      itp('redirects automatically to third party when lastFailedChallengeFactorData is null', function() {
+        spyOn(SharedUtil, 'redirect');
```

Comment:
Similar to comment above - please use Jest objects instead of Jasmine:

```suggestion
        jest.spyOn(SharedUtil, 'redirect');
```

Reply:
@jmelberg-okta can you resolve this comment please.